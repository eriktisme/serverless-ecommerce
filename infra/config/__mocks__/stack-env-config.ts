import { StackConfiguration } from '../types'

export const StackEnvConfiguration = (): StackConfiguration => ({
  project: 'test-project',
  stage: 'jest',
  env: {
    account: '000000000000',
    region: 'eu-west-1',
  },
  domain: {
    domain: 'scaling.test',
  },
})
