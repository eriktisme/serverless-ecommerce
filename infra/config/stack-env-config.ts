import { DevConfig } from './env-config-dev'
import { IntConfig } from './env-config-int'
import { ProdConfig } from './env-config-prod'
import { StackConfiguration } from './types'

interface ConfigurationsContext {
  [key: string]: StackConfiguration
}

const configurationsContext: ConfigurationsContext = {
  int: IntConfig,
  dev: DevConfig,
  prod: ProdConfig,
}

export const StackEnvConfiguration = (
  configContext: string
): StackConfiguration => {
  if (
    !Object.prototype.hasOwnProperty.call(configurationsContext, configContext)
  ) {
    throw new Error(
      'Missing configuration context. Provide a configuration context using eg: `--context config=int`'
    )
  }

  return configurationsContext[configContext]
}
