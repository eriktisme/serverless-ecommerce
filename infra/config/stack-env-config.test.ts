import { StackEnvConfiguration } from './stack-env-config'

describe('Stack environment configuration', () => {
  it('throws an error when the configuration context does not match predefined contexts', () => {
    expect(() => {
      StackEnvConfiguration('undefined')
    }).toThrow()
  })

  it('returns int stack configuration given context=int', () => {
    const stackConfig = StackEnvConfiguration('int')

    expect(stackConfig).toMatchSnapshot()
  })

  it('returns int stack configuration given context=dev', () => {
    const stackConfig = StackEnvConfiguration('dev')

    expect(stackConfig).toMatchSnapshot()
  })

  it('returns prod stack configuration given context=prod', () => {
    const stackConfig = StackEnvConfiguration('prod')

    expect(stackConfig).toMatchSnapshot()
  })
})
