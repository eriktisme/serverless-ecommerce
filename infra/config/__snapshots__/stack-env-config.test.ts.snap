// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stack environment configuration returns int stack configuration given context=dev 1`] = `
Object {
  "domain": Object {
    "domain": "scaling.cloud",
  },
  "env": Object {
    "account": "096007086684",
    "region": "eu-west-1",
  },
  "project": "serverless-ecommerce",
  "stage": "dev",
}
`;

exports[`Stack environment configuration returns int stack configuration given context=int 1`] = `
Object {
  "domain": Object {
    "domain": "scaling.cloud",
  },
  "env": Object {
    "account": "096007086684",
    "region": "eu-west-1",
  },
  "project": "serverless-ecommerce",
  "stage": "int",
}
`;

exports[`Stack environment configuration returns prod stack configuration given context=prod 1`] = `
Object {
  "domain": Object {
    "domain": "scaling.cloud",
  },
  "env": Object {
    "account": "096007086684",
    "region": "eu-west-1",
  },
  "project": "serverless-ecommerce",
  "stage": "prod",
}
`;
